src/plugins/discover/public/application/view_components/canvas/discover_table_app.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useState, useEffect } from 'react';
+import { EuiPage, EuiPageBody, EuiFlexGroup, EuiFlexItem, EuiPageContent } from '@elastic/eui';
+import { DataGridTable } from '../../components/data_grid/data_grid_table';
+
+export const DiscoverTableApplication = ({ data$, indexPattern, savedSearch, services }) => {
+  const [fetchState, setFetchState] = useState<any>({
+    status: data$.getValue().status,
+    fetchCounter: 0,
+    fieldCounts: {},
+    rows: [],
+  });
+
+  const { rows } = fetchState;
+
+  useEffect(() => {
+    const subscription = data$.subscribe((next) => {
+      if (
+        (next.status && next.status !== fetchState.status) ||
+        (next.rows && next.rows !== fetchState.rows)
+      ) {
+        setFetchState({ ...fetchState, ...next });
+      }
+    });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [data$, fetchState]);
+
+  // ToDo: implement columns, onAddColumn, onRemoveColumn, onMoveColumn, onSetColumns using config, indexPattern, appState
```

Comment:
most of these are already done in my side panel PR so i'd hold off implementing this until then :)